bull/email-queue/consumer.js

```diff
@@ -1,49 +1,79 @@
-const emailtemp = require("../../utils/email");
 const config = require("../../utils/bullconfig");
-const nodemailer = require("nodemailer");
 const EmailQueue = config.getNewQueue("email-queue");
 const winston = require("winston");
 const logger = winston.loggers.get("defaultLogger");
-
 require("dotenv").config();
+const { Mwn } = require("mwn");
 
-const transporter = nodemailer.createTransport({
-  service: process.env.service,
-  auth: {
-    user: process.env.email,
-    pass: process.env.password,
-  },
-});
-
-EmailQueue.on("active", (job, jobPromise) => {});
+/* 
+Mediawiki Email API DOCS - https://www.mediawiki.org/wiki/API:Emailuser#JavaScript
+MWN TOOLFORGE PACKAGE DOCS -https://github.com/siddharthvp/mwn
+*/
 
-EmailQueue.on("completed", (job, result) => {});
+async function mediawikiEmail(username, title, trueURI, success) {
+  try {
+    const bot = await Mwn.init({
+      apiUrl: process.env.EMAIL_SOURCE_URL,
+      username: process.env.EMAIL_BOT_USERNAME,
+      password: process.env.EMAIL_BOT_PASSWORD,
+      // Set your user agent (required for WMF wikis, see https://meta.wikimedia.org/wiki/User-Agent_policy):
+      userAgent: "BUB2/1.0 (https://bub2.toolforge.org)",
+      // Set default parameters to be sent to be included in every API request
+      defaultParams: {
+        assert: "user", // ensure we're logged in
+      },
+    });
 
-EmailQueue.process((job, done) => {
-  if (job.data.email != "") {
-    const mailOptions = {
-      from: process.env.email, // sender address
-      to: job.data.email, // list of receivers
-      subject: job.data.success
-        ? 'BUB File Upload - "Successful"'
-        : 'BUB File Upload - "Error"', // Subject line
-      html: emailtemp.emailtemplate(
-        job.data.title,
-        job.data.success,
-        job.data.trueURI
-      ), // plain text body
-    };
+    const csrf_token = bot.getCsrfToken();
```

Comment:
[https://mwn.toolforge.org/docs/api/classes/mwn.html#getcsrftoken](getcsrftoken) from the doc returns promise. Add `await` here.